tests/replication2_ssl.erl

```diff
@@ -14,221 +49,153 @@ confirm() ->
 
-    SSLConfig9 = [
-        {riak_repl,
-            [
-                {fullsync_on_connect, false},
-                {fullsync_interval, disabled}
-            ]},
-        {riak_core,
-            [
-                {ssl_enabled, true},
-                {peer_common_name_acl, ["*.basho.com"]},
-                {certfile, filename:join([CertDir,
-                    "wildcard.basho.com/cert.pem"])},
-                {keyfile, filename:join([CertDir,
-                    "wildcard.basho.com/key.pem"])},
-                {cacertdir, filename:join([CertDir,
-                    "wildcard.basho.com/cacerts.pem"])}
-            ]}
+    ConfRepl = {riak_repl,
+        [{fullsync_on_connect, false}, {fullsync_interval, disabled}]},
+
+    ConfTcpBasic = [ConfRepl, {riak_core, [{ssl_enabled, false}]}],
+
+    %%
+    %% !!! IMPORTANT !!!
+    %% Properties added to node configurations CANNOT currently be removed,
+    %% only overwritten.  As such, configurations that include ACLs MUST come
+    %% after ALL non-ACL configurations!  This has been learned the hard way :(
+    %%
+
+    %%
+    %% Connection Test descriptors
+    %% Each is a tuple: {Description, Node1Config, Node2Config, Should Pass}
+    %%
+    SslConnTests = [
+        {"identical certificate CN is disallowed",
+            [ConfRepl, {riak_core, [{ssl_enabled, true}
+            ] ++ CIsite1#ci.ssl }],
+            [ConfRepl, {riak_core, [{ssl_enabled, true}
+            ] ++ CIsite1#ci.ssl }],
+            false},
+        {"non-SSL peer fails",
+            [ConfRepl, {riak_core, [{ssl_enabled, true}
+            ] ++ CIsite1#ci.ssl }],
+            ConfTcpBasic,
+            false},
+        {"non-SSL local fails",
+            ConfTcpBasic,
+            [ConfRepl, {riak_core, [{ssl_enabled, true}
+            ] ++ CIsite2#ci.ssl }],
+            false},
+        {"basic SSL connectivity",
+            [ConfRepl, {riak_core, [{ssl_enabled, true}
+            ] ++ CIsite1#ci.ssl }],
+            [ConfRepl, {riak_core, [{ssl_enabled, true}
+            ] ++ CIsite2#ci.ssl }],
+            true},
+        {"SSL connectivity with intermediate CA",
+            [ConfRepl, {riak_core, [{ssl_enabled, true}
+            ] ++ CIsite1#ci.ssl }],
+            [ConfRepl, {riak_core, [{ssl_enabled, true}
+            ] ++ CIsite3#ci.ssl }],
+            true},
+        {"wildcard certifictes on both ends",
+            [ConfRepl, {riak_core, [{ssl_enabled, true}
+            ] ++ CIsiteWC#ci.ssl }],
+            [ConfRepl, {riak_core, [{ssl_enabled, true}
+            ] ++ CIsiteWC#ci.ssl }],
+            true},
+        {"wildcard certifictes on one end",
+            [ConfRepl, {riak_core, [{ssl_enabled, true}
+            ] ++ CIsiteWC#ci.ssl }],
+            [ConfRepl, {riak_core, [{ssl_enabled, true}
+            ] ++ CIsite3#ci.ssl }],
+            true},
+        %% first use of ssl_depth, all subsequent tests must specify it
+        {"disallowing intermediate CA setting",
```

Comment:
Please tweak the description on this or the next test, confusing two see two nearly identical tests with identical descriptions but one is intended to fail and one passes.